internal/provider/resource_vault_cluster.go

```diff
@@ -1031,6 +1110,45 @@ func flattenObservabilityConfig(config *vaultmodels.HashicorpCloudVault20201125O
 			}
 		}
 	}
+
+	if cloudwatch := config.Cloudwatch; cloudwatch != nil {
+		configMap["cloudwatch_access_key_id"] = cloudwatch.AccessKeyID
+		configMap["cloudwatch_region"] = cloudwatch.Region
+		// ensure we only set properties that are defined in metrics/audit-logs streaming
+		if propertyName == "metrics_config" {
+			// Namespace is only used for streaming metrics
+			configMap["cloudwatch_namespace"] = cloudwatch.Namespace
+		} else {
+			// Stream name and group name are only used for streaming audit-logs
+			configMap["cloudwatch_stream_name"] = cloudwatch.StreamName
+			configMap["cloudwatch_group_name"] = cloudwatch.GroupName
+		}
+		// Since the API return this sensitive fields as redacted, we don't update it on the config in this situations
+		if cloudwatch.SecretAccessKey != "redacted" {
+			configMap["cloudwatch_secret_access_key"] = cloudwatch.SecretAccessKey
+		} else {
+			if configParam, ok := d.GetOk(propertyName); ok && len(configParam.([]interface{})) > 0 {
+				config := configParam.([]interface{})[0].(map[string]interface{})
+				configMap["cloudwatch_secret_access_key"] = config["cloudwatch_secret_access_key"].(string)
+			}
+		}
+
+		if elasticsearch := config.Elasticsearch; elasticsearch != nil {
+			configMap["elasticsearch_endpoint"] = elasticsearch.Endpoint
+			configMap["elasticsearch_user"] = elasticsearch.User
+
```

Comment:
I noticed that there was a field called `dataset` on the `config.ElasticSearch` struct. Does that go here as well? Should it only be set for streaming metrics or audit logs?